### DISCUSSION TITLE
SaveFileDialog - turn off overwrite confirmation?

### DISCUSSION THREAD

**Discussion by Zilkode** [AvaloniaUI/Avalonia#5756] [Q&A] [answered]
When using SaveFileDialog, I want to implement a dialog of my own to give the user few choices.
The above control forces the user to choose between replacing the file or choose another filename.

In my case, I want to let the user "resume".
Is there any way around? maybe another picker i am missing? or should it be done as a new feature?

Thanks.

**Comment by maxkatz6** [comment, marked as answer]
If you want your own file dialog, you can implement this interface [ISystemDialogImpl](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Platform/ISystemDialogImpl.cs) and setup it as here https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Dialogs/ManagedFileDialogExtensions.cs#L68

Or you can completely ignore Avalonia side and use your own implementation directly.

**Reply by Zilkode** [reply]
Thank you.
